src/Library/demos/Menu/main.blp

```diff
@@ -0,0 +1,159 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.StatusPage demo {
+  title: _("Menu");
+  description: _("Display structured menus with items, sections and submenus");
+
+  Box {
+    halign: center;
+    orientation: vertical;
+
+    Box {
+      orientation: vertical;
+      spacing: 42;
+
+      Adw.PreferencesGroup {
+        title: _("Primary and Secondary Menus");
+        description: _("Used with controls such as menu buttons");
+
+        MenuButton {
+          halign: center;
+          label: _("Menu Button");
+          menu-model: demo_menu;
+        }
+      }
+
+      Adw.PreferencesGroup {
+        title: _("Context Menus");
+        description: _("Menus that popup on right-clicking a widget");
+
+        Adw.Clamp {
+          maximum-size: 500;
+
+          Frame {
+            Label label {
+              margin-top: 6;
+              margin-bottom: 6;
+              margin-start: 6;
+              margin-end: 6;
+              wrap: true;
+              wrap-mode: char;
+              selectable: true;
+              label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Vel elit scelerisque mauris pellentesque pulvinar. Molestie nunc non blandit massa enim nec dui nunc.";
+              extra-menu: context_menu;
+            }
+          }
+        }
+      }
+
+      Box {
+        halign: center;
+
+        LinkButton {
+          label: "GJS Guide";
+          uri: "https://gjs.guide/guides/gio/actions-and-menus.html#gmenu";
+        }
+
+        LinkButton {
+          label: "Human Interface Guidelines";
+          uri: "https://developer.gnome.org/hig/patterns/controls/menus.html";
+        }
+      }
+    }
+  }
+}
+
+menu demo_menu {
+  item {
+    label: _("New File");
+  }
+
+  section {
+    item {
+      label: _("New Terminal");
+    }
+
+    item {
+      label: _("New Build Terminal");
+    }
+
+    item {
+      label: _("New Runtime Terminal");
+    }
+  }
+
+  section {
+    label: _("Settings");
+
+    submenu {
+      label: _("Appearance");
+
+      item {
+        label: _("Force System Default");
+      }
+
+      item {
+        label: _("Force Light Mode");
+      }
+
+      item {
+        label: _("Force Dark Mode");
+      }
+    }
+
+    submenu {
+      label: _("Accessibility");
+
+      item {
+        label: _("High Contrast");
+      }
+
+      section {
+        label: _("Text Direction");
+
+        item {
+          label: _("Left-to-Right");
+        }
+
+        item {
+          label: _("Right-to-Left");
+        }
+      }
+    }
+  }
+}
+
+menu context_menu {
+  item {
+    label: _("Italics");
+    action: "text.italic";
+  }
+
+  item {
+    label: _("Bold");
+    action: "text.bold";
+  }
+
+  submenu {
+    label: _("Font Color");
+
+    item {
+      label: _("White");
+      action: "text.color";
+      target: "white";
```

Comment:
```suggestion
      label: _("Black");
      action: "text.color";
      target: "black";
```

Let's make the default `black`. The default makes it go nearly invisible, and I think that's a bit confusing.

Reply:
Sorry for the late response 😅 

But I was running Workbench in dark mode so I completely forgot that white text would go invisible in light mode, maybe we could choose a color that's neither black nor white?

Maybe gray?

![image](https://github.com/sonnyp/Workbench/assets/58233418/e88a27b8-551a-4bf3-a352-840b1f621568)
![image](https://github.com/sonnyp/Workbench/assets/58233418/c698db2f-d292-4295-a058-ec7148734494)

We could choose any other random color

Reply:
How about just `green`? I hadn't thought of dark mode myself, so red/green/blue seems safe and legible for all variants.